### ISSUE TITLE
New Hook to allow alternative order number solutions.

### ISSUE THREAD

**Issue by PloughGuy** [cubecart/v6#1867]
# Motivation
My existing store uses sequential order numbers which I wish to maintain. 
No doubt other users of CubeCart have custom order number requirements other than the default.
This proposal allows substitution of custom order numbers for the default <date>-<time>-<random 4 digits> solution.

# Starting point
In the interim, I will be adding a personal hook to my store in order_class.  I know you all know way more about this than I do, but this will be my solution until something more formal comes along.

#And another thing
Can we fix the spelling of "explanatory"?

Thank you for considering this enhancement.
Russ

	public function createOrderId($return = false, $set_basket = true) {
		// Self-explainitory really...
		
		$this->_order_id = "";
		****
		if ($this->_order_id == "")
			$this->_order_id = date('ymd-His-').rand(1000, 9999);
		
		if ($set_basket) {
			$this->_basket['cart_order_id'] = $this->_order_id;
			/* fix for admin generated orders */
			if (method_exists($GLOBALS['cart'], 'save')) {
				$GLOBALS['cart']->save();
			}
		}
		return ($return) ? $this->_order_id : true;
	}

**Comment by abrookbanks** [comment]
Be careful because there are many other parts of CubeCart that expect certain order formatting!!

See below (There may be more)
https://github.com/cubecart/v6/blob/master/admin/sources/orders.transactions.inc.php#L40
https://github.com/cubecart/v6/blob/master/classes/order.class.php#L780

I hate to throw a negative spin on this because I want to see this functionality but you are likely to run into a number of problems. I took a different approach on this by adding a new auto incremented column to the orders table to pull an alternate order ID into the dataset but this too didn't come without problems and I reverted all my code.

When this style was originally chosen (about 15 years a go) it was a bit of an Amazon clone. The last item I ordered from Amazon had order 202-9372677-5745949 which doesn't appear to be incremented.

Maybe the best solution is to go back to adding an auto incremented value which just shows on "official purposes" like invoices and admin order exports. The order number generated by CubeCart could be changed to have a different terminology such as "order reference".
